packages/vite-plugin-cloudflare/src/miniflare-options.ts

```diff
@@ -354,6 +355,52 @@ export async function getDevMiniflareOptions(config: {
 				},
 			},
 		},
+		{
+			name: ASSET_PROXY_WORKER_NAME,
+			compatibilityDate: ASSET_WORKERS_COMPATIBILITY_DATE,
+			modulesRoot: miniflareModulesRoot,
+			modules: [
+				{
+					type: "ESModule",
+					path: path.join(miniflareModulesRoot, "asset-proxy-worker.js"),
+					contents: /*javascript*/ `
+						import { WorkerEntrypoint } from 'cloudflare:workers';
+						export default class RPCProxyWorker extends WorkerEntrypoint {
+							async fetch(request) {
+								return this.env.FETCHER.fetch(request);
+							}
+
+							tail(event) {
+								// Temporary workaround: the tail event is not serializable,
+								// so we are serializing it to JSON and parsing it back to make it transferable.
+								// This loses non-serializable data, but allows us to forward basic info
+								// to the target worker until native support is available.
+							    return this.env.RPC_WORKER.tail(JSON.parse(JSON.stringify(event)));
```

Comment:
We are expecting native support from the runtime team in a couple weeks: [EW-9311](https://jira.cfdata.org/browse/EW-9311)